### PR TITLE
machine/itsybitsy-nrf52840: add support for Adafruit Itsybitsy nrf52840

### PR DESCRIPTION
**Pull request by deadprogram**
This PR adds support for Adafruit Itsybitsy nrf52840 Express board.

### CODE REVIEW

**Comment by sago35** [comment]
product page: https://www.adafruit.com/product/4481
nRF52840 datasheet: https://infocenter.nordicsemi.com/pdf/nRF52840_PS_v1.0.pdf
PinOut: https://learn.adafruit.com/adafruit-itsybitsy-nrf52840-express/pinouts
Schematics: https://learn.adafruit.com/assets/87158

File: src/machine/board_itsybitsy-nrf52840.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+// USB CDC identifiers
+const (
+	usb_STRING_PRODUCT      = "Adafruit ItsyBitsy nRF52840 Express"
+	usb_STRING_MANUFACTURER = "Adafruit"
+)
+
+var (
+	usb_VID uint16 = 0x239A
+	usb_PID uint16 = 0x0052
```
Comment (sago35) [inline src/machine/board_itsybitsy-nrf52840.go]: Which is correct, 0x0051 (pid.1) or 0x0052 (pid.2)?
Or does it matter either way?

feather-nrf82540 has 0x0029 (pid.1) and 0x002A (pid.2) in boards.txt There are several, but 0x0029 (pid.1) has been selected.

https://github.com/adafruit/Adafruit_nRF52_Arduino/blob/master/boards.txt#L180

File: src/machine/board_itsybitsy-nrf52840.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+const HasLowFrequencyCrystal = true
+
+// GPIO Pins
+const (
+	D0  = P0_25 // UART TX
+	D1  = P0_24 // UART RX
+	D2  = P1_02
+	D3  = P0_06 // LED1
+	D4  = P0_20 // Button
```
Comment (sago35) [inline src/machine/board_itsybitsy-nrf52840.go]: Perhaps P0_29 is correct.
d4 
https://cdn-learn.adafruit.com/assets/assets/000/087/158/original/adafruit_products_schem.png?1579387035

File: README.md [hunk 3]
```diff
@@ -53,6 +53,7 @@ The following 38 microcontroller boards are currently supported:
 * [Adafruit Feather nRF52840 Express](https://www.adafruit.com/product/4062)
 * [Adafruit ItsyBitsy M0](https://www.adafruit.com/product/3727)
 * [Adafruit ItsyBitsy M4](https://www.adafruit.com/product/3800)
+* [Adafruit ItsyBitsy nrf52840](https://www.adafruit.com/product/4481)
```
Comment (sago35) [inline README.md]: I think the same notation as the Adafruit Feather nRF52840 Express should be used. before : nrf52840

before : nrf52840
after : nRF52840

**Review by sago35 — COMMENTED** [review]
I checked the following.

* Makefile.
  * Make sure I can build
* README.md
  * Check the product page & URL
* src/machine/board_itsybitsy-nrf52840.go
  * Make sure it's the same as the following
  * https://learn.adafruit.com/adafruit-itsybitsy-nrf52840-express/pinouts
  * https://learn.adafruit.com/assets/87158
  * https://github.com/adafruit/Adafruit_nRF52_Arduino/blob/master/variants/itsybitsy_nrf52840_express/variant.cpp
* targets/itsybitsy-nrf52840.json
  * targets/feather-nrf52840.json and targets/ Compare with circuitplay-bluefruit.json

File: src/machine/board_itsybitsy-nrf52840.go (hunk 1 above)
Comment (deadprogram) [reply src/machine/board_itsybitsy-nrf52840.go]: I think it should be `0x8051` based on that link:

https://github.com/adafruit/Adafruit_nRF52_Arduino/blob/master/boards.txt#L210

Comment (bgould) [reply src/machine/board_itsybitsy-nrf52840.go]: Hmm ok good find, I will need to double check the feather IDs I suppose

File: src/machine/board_itsybitsy-nrf52840.go (hunk 2 above)
Comment (aykevl) [reply src/machine/board_itsybitsy-nrf52840.go]: Indeed: https://github.com/adafruit/Adafruit_nRF52_Arduino/blob/master/variants/itsybitsy_nrf52840_express/variant.cpp#L33

File: README.md (hunk 3 above)
Comment (deadprogram) [reply README.md]: Updated.

File: src/machine/board_itsybitsy-nrf52840.go (hunk 2 above)
Comment (deadprogram) [reply src/machine/board_itsybitsy-nrf52840.go]: Changed.

**Review by aykevl — APPROVED** [review]
Looks good to me (but perhaps others want to take a look).

File: src/machine/board_itsybitsy-nrf52840.go (hunk 1 above)
Comment (deadprogram) [reply src/machine/board_itsybitsy-nrf52840.go]: Updated.

**Comment by deadprogram** [comment]
OK, now squash/merging. Thank you everyone for the help on this!

**Pull request merged**